packages/cli/src/lib/project/dotFolder.ts

```diff
@@ -0,0 +1,53 @@
+import {
+  createFileSync,
+  existsSync,
+  pathExistsSync,
+  writeJSONSync,
+} from 'fs-extra';
+import {join} from 'path';
+
+export class DotFolder {
+  public constructor(public owner: string) {
+    this.ensureFolderExists();
+  }
+
+  public static get hiddenFolderName() {
+    return '.coveo';
+  }
+
+  public get path() {
+    return join(this.owner.toString(), DotFolder.hiddenFolderName);
+  }
+
+  private ensureFolderExists() {
+    const exists = pathExistsSync(this.path);
+    if (!exists) {
+      createFileSync(this.path);
+    }
+  }
+}
+
+export class DotFolderConfig {
+  public constructor(public ownerFolder: DotFolder) {
+    this.ensureFileExists();
+  }
+
+  public static get configName() {
+    return 'config.json';
+  }
+
+  private get defaultConfig() {
+    return {
+      version: 1,
+      organization: '', // TODO: Needs to use that "local" config for all :config:* command that would rely on it.
+    };
+  }
+
+  private ensureFileExists() {
+    const path = join(this.ownerFolder.path, DotFolderConfig.configName);
+    if (!existsSync(path)) {
```

Comment:
I think `writeJSONSync` already checks if the file already exists and creates it if that is not the case.